arangod/IResearch/IResearchFilterFactory.cpp

```diff
@@ -1336,6 +1311,66 @@ bool fromFuncPhrase(
     }
   }
 
+  // ...........................................................................
+  // 2nd argument defines a value
+  // ...........................................................................
+
+  auto const* valueArg = args.getMemberUnchecked(1);
+
+  if (!valueArg) {
+    LOG_TOPIC(WARN, arangodb::iresearch::IResearchFeature::IRESEARCH)
+      << "'PHRASE' AQL function: 2nd argument is invalid";
```

Comment:
In general no warnings should be logged in AQL functions if the caller somehow managed to pass wrong input parameters into the function.
In the general case the function input parameter values are supplied by a human, and they will not see these warnings as part of the query result, or when they run the queries from the web interface.
Instead, a warning should be registered with the AQL Query object (Query::registerWarning).
That also applies for all following cases.

Reply:
Is there an example of this anywhere?
So far all the IResearch related code always logs warnings if an error code is returned, makes it much simpler to debug the code

Reply:
Yes, several. I suggest looking here as it will contain most examples: 
```$ grep -ri "registerWarning" arangod/Aql/Functions.cpp | wc -l
58
```

Reply:
what is '58'?

Reply:
sorry, wrong formatting. Now it should be better.